apps/web/src/components/WalletConnect/SessionProposalModal.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  Box,
+  Card,
+  FormControl,
+  FormErrorMessage,
+  HStack,
+  ModalContent,
+  Text,
+  VStack,
+  useToast,
+} from "@chakra-ui/react";
+import { type WalletKitTypes } from "@reown/walletkit";
+import { useDynamicModalContext } from "@umami/components";
+import { useAsyncActionHandler, useGetImplicitAccount, walletKit } from "@umami/state";
+import { buildApprovedNamespaces, getSdkError } from "@walletconnect/utils";
+import { useState } from "react";
+import { FormProvider, useForm } from "react-hook-form";
+
+import RequestModal from "./RequestModal";
+import { CheckmarkIcon, CloseIcon } from "../../assets/icons";
+import { OwnedImplicitAccountsAutocomplete } from "../AddressAutocomplete";
+
+/**
+ * Finds the overlap between the chains supported by the wallet and the chains requested by the dApp.
+ * We take the first required chain as Umami supports connection for only one chain per request.
+ */
+function getChainToConfirm(required: (string | string[] | undefined)[]) {
+  // take the first required chain which is supported by the wallet
+  for (const chain of required.flat()) {
+    if (chain && ["tezos:mainnet", "tezos:ghostnet"].includes(chain)) {
+      return chain;
+    }
+  }
+  return undefined;
+}
+
+export default function SessionProposalModal({
+  proposal,
+}: {
+  proposal: WalletKitTypes.SessionProposal;
+}) {
+  console.log("SessionProposalModal", proposal);
+  const getAccount = useGetImplicitAccount();
+  const toast = useToast();
+
+  const { onClose } = useDynamicModalContext();
+  const { handleAsyncAction } = useAsyncActionHandler();
+
+  const [isLoadingApprove, setIsLoadingApprove] = useState(false);
+  const [isLoadingReject, setIsLoadingReject] = useState(false);
+  const form = useForm<{ address: string }>({
+    mode: "onBlur",
+  });
+  const {
+    getValues,
+    formState: { errors, isValid },
+  } = form;
+
+  const required = Object.entries(proposal.params.requiredNamespaces).map(([key, values]) =>
+    key.includes(":") ? key : values.chains
+  );
+
+  if (required.length !== 1) {
+    console.error("Expected only one required chain", required);
+  }
+
+  const chain = getChainToConfirm(required);
+  if (!chain) {
+    console.error("No supported chain found", required);
+  }
+
+  const onApprove = () =>
+    handleAsyncAction(async () => {
+      const account = getAccount(getValues().address);
+
+      try {
+        if (!chain) {
+          throw new Error("No supported chain found");
+        }
+        const namespaces = buildApprovedNamespaces({
+          proposal: proposal.params,
+          supportedNamespaces: {
+            tezos: {
+              chains: [chain],
+              methods: ["tezos_getAccounts", "tezos_sign", "tezos_send"],
```

Comment:
Another product question: maybe we can allow user to select which actions they want to allow for the given dApp?

Reply:
maybe, but that's a new feature which nobody has requested. I suggest to leave it until somebody explicitly wants it

Reply:
Agree, definitely not needed for this PR

But could be good to discuss it with Aswin and maybe create add a backlog task for later (so that it's not get lost)